config.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
```

Comment:
Is it in the scope of the PR to start using the v2 version of `go-utils` and `go-steputils`? I don't want to block this PR, so it's fine for me as-is, but most of these imports have v2 equivalents that make testing easier.

Reply:
We have a long term goal of moving to the v2 equivalents across the board.  If we're taking a look at this step at this detail, it feels appropriate to update now and save us some work later.

Reply:
The goal of this PR is mainly to move code around to the more organized format of the ideal step architecture and handle dependency injection along with unit tests. Do we agree that package upgrades should also be done as conversion to ideal step architecture?

Reply:
Unless there are some massive breaking changes in the v2 version of a utility, I would say let's make the change

Reply:
Can we sync on how to accomplish this? I'm unable to find a migration guide and don't know what to do in scenarios where a package isn't available in v2 (e.g.: `github.com/bitrise-io/go-utils/retry`).